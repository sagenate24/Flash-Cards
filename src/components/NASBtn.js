import React from 'react';
import {
  View,
  Text,
  Platform,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import { white } from '../utils/colors';

// The name is formed from my name just in case projects
// That need to be shared between developers dont have matching names like Button
export default function NASBtn({
  children, onPress, disabled, tintColor,
}) {
  return (
    <View style={styles.buttonContainer}>
      <TouchableOpacity
        disabled={disabled}
        onPress={onPress}
        style={[styles.button, tintColor, disabled && { opacity: 0.8 }]}
      >
        <Text style={styles.btnText}>{children}</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  buttonContainer: {
    paddingTop: 10,
    paddingLeft: 20,
    paddingRight: 20,
  },
  button: {
    padding: 10,
    borderRadius: Platform.OS === 'ios' ? 0 : 2,
  },
  btnText: {
    color: white,
    fontSize: Platform.OS === 'ios' ? 22 : 18,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
