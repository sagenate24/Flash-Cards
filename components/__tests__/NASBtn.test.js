import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NASBtn from '../NASBtn';

test('NASBtn snapshot', () => {
  const snap = renderer.create(
    <NASBtn disabled={false} onPress={jest.fn()}>test this text</NASBtn>
  ).toJSON();

  expect(snap).toMatchSnapshot();
})