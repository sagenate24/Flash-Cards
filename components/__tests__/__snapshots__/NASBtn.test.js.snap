// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<View
  style={
    Object {
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 10,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 0,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 22,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      test this text
    </Text>
  </View>
</View>
`;
